app/views/editor/level/thangs/AddThangsView.coffee

```diff
@@ -23,8 +25,23 @@ module.exports = class AddThangsView extends CocoView
     super options
     @world = options.world
 
+    @thangTypes = new Backbone.Collection()
+    thangTypeCollection = new ThangTypeSearchCollection([])
+    thangTypeCollection.fetch({data: {limit: PAGE_SIZE}})
+    thangTypeCollection.skip = 0
     # should load depended-on Components, too
-    @thangTypes = @supermodel.loadCollection(new ThangTypeSearchCollection(), 'thangs').model
+    @supermodel.loadCollection(thangTypeCollection, 'thangs')
+    @listenTo thangTypeCollection, 'sync', @onThangCollectionSynced
```

Comment:
If this should only trigger once, then maybe this should be `listenToOnce` which cleans up the listener after firing?
Reference: http://backbonejs.org/#Events-listenTo

Reply:
Not necessary it will be once, what if there are more than 2000 or PAGE_SIZE will be less than 1000.

Reply:
I definitely agree.